packages/core/core-flows/src/order/workflows/claim/update-claim-add-item.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  OrderChangeActionDTO,
+  OrderChangeDTO,
+  OrderClaimDTO,
+  OrderDTO,
+  OrderWorkflow,
+} from "@medusajs/types"
+import { ChangeActionType, OrderChangeStatus } from "@medusajs/utils"
+import {
+  WorkflowData,
+  createStep,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { useRemoteQueryStep } from "../../../common"
+import {
+  previewOrderChangeStep,
+  updateOrderChangeActionsStep,
+} from "../../steps"
+import {
+  throwIfIsCancelled,
+  throwIfOrderChangeIsNotActive,
+} from "../../utils/order-validation"
+
+const validationStep = createStep(
+  "update-claim-add-item-validation",
+  async function (
+    {
+      order,
+      orderChange,
+      orderClaim,
+      input,
+    }: {
+      order: OrderDTO
+      orderClaim: OrderClaimDTO
+      orderChange: OrderChangeDTO
+      input: OrderWorkflow.UpdateClaimAddNewItemWorkflowInput
+    },
+    context
+  ) {
+    throwIfIsCancelled(order, "Order")
+    throwIfIsCancelled(orderClaim, "Claim")
+    throwIfOrderChangeIsNotActive({ orderChange })
+
+    const associatedAction = (orderChange.actions ?? []).find(
+      (a) => a.id === input.action_id
+    ) as OrderChangeActionDTO
+
+    if (!associatedAction) {
+      throw new Error(
+        `No request to add item for claim ${input.claim_id} in order change ${orderChange.id}`
+      )
+    } else if (associatedAction.action !== ChangeActionType.ITEM_ADD) {
+      throw new Error(`Action ${associatedAction.id} is not adding an item`)
+    }
+  }
+)
+
+export const updateClaimAddItemWorkflowId = "update-claim-add-item"
+export const updateClaimAddItemWorkflow = createWorkflow(
```

Comment:
question: Maybe this is obvious and I am just missing a bit of context, but how is this workflow different from `updateClaimItemWorkflow`? It looks like only the validation differs. Is that intentional?

Reply:
It is very likely the same.
At this point all these post purchase flows have much more repetition than I'm comfortable maintaining. After a cleanup I expect more generic flows that can be reused and many ones renamed or removed.